app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -1024,6 +1025,9 @@ private void loadForm() {
                             instanceState);
                     formId = instanceAndStatus.first;
                     instanceIsReadOnly = instanceAndStatus.second;
+
+                    // only retrieve a potentially stored form index when loading an existing form record
+                    serializedFormIndex = HiddenPreferences.getInterruptedFormIndex();
```

Comment:
nit: probably cleaner to directly access it in `AndroidFormController` instead of passing it through the loader task to controller ?

Reply:
I think there are somewhat competing design constraints. Ideally, I think it's best for Android OS lifecycle type code to be isolated into Activities, or other clearly-coupled-to-the-OS locations, so it does make sense to me that the root entry point for actually loading the form index should be in the activity. 

In thinking about where to put these call (and what format to put them in) my two main priorities would be:

* I think it's good for the retrieval code, the storage code, and the serialization/deserialization to all be as compartmentalized as possible so there's little ambiguity about whether a future change has completely covered them
* I think we should try and minimize state uncertainty. As written one big potential point of confusion is that there's lots of uncertainty about this return object. Does it exist at all? Is it serialized correctly? Is it valid for the current form? It worries me that uncertainty is passed through later to the loader task, since it increases the number of touchpoints where a reader can't be certain about what might happen with it.